File: Unit 1 Fundamentals of programming/Fundamentals Topic 6 Files and exception handling/Topic 6 Python Programs/H6 School match scores.py
#AQA AS H6 School match scores

#subroutine to display menu
def menu():
    print("Select from the following options: \n")
    print("1 Enter and save new data")
    print("2 Load and display data")
    print("3 Quit")
    response = input("\nEnter your selection: ")
    while response<"1" or response >"3":
        response = input("Error... enter a number between 1 and 3: ")
        
    return response

#subroutine to enter new data
def getData(school, wins):
    for index in range(3):
       valid = False
       school[index]=input("Enter school: ")
       while not valid:
           try:
               wins[index] = input ("Enter wins: ")
               numwins = int(wins[index])    #test for integer
               while numwins < 0 or numwins>20:                  
                   wins[index] = input ("Error... re-enter wins: ")
                   numwins = int(wins[index])   #test for integer          
               valid = True
           except:
               print("Please enter an integer")
            #end except
        #endwhile
    #endfor
#endsub
               
#subroutine to save data to disk
def save(school, wins, filename):
    matchFile = open(filename,"w")
    for index in range(3):
       matchFile.write(school[index]+"," + wins[index] + "\n")
    print("\ndata saved\n")
    matchFile.close()

#subroutine to load records into an array and display data
def loadAndDisplay(school, wins,filename):
    matchFile = open(filename,"r")
    
    for index in range(3):
        schoolRec = matchFile.readline()
        field = schoolRec.split(",")
        school[index] = field[0]
        wins[index] = field[1]
        print ("School: ",school[index], "  Wins: ",wins[index])
    matchFile.close()



#************************* MAIN PROGRAM  ****************************               
school =["","",""]
wins = [0,0,0]
filename = "teamScores.txt"
choice = menu()
while choice >="1" and choice <="3":
    if choice == "1":
        getData(school,wins)
        save(school, wins, filename)
    elif choice == "2":
        loadAndDisplay(school, wins, filename)
    else:
        print("You have chosen to end the program")
        break   #exit while loop
    choice = menu()
    #endif
#endwhile
#end of program

File: A-Level/Unit 1 Fundamentals of programming/Fundamentals Topic 6 Files and exception handling/Topic 6 VB Programs/H6 School match scores.vb
Module Module1
    Dim maxArray As Integer = 2 'global variable allows you to easily change array size for testing
    Sub Main()
        'H6 School match scores

        Dim schools(maxArray) As String
        Dim wins(maxArray) As Integer
        Dim filename As String = "U:\teamScores.txt"
        Dim choice As Integer = menu()
        While choice <> 6
            Select Case choice
                Case 1
                    load(schools, wins, filename)
                Case 2
                    getData(schools, wins)
                Case 3
                    save(schools, wins, filename)
                Case 4
                    displayData(schools, wins)
                Case 5
                    menu()
                Case 6
                    Console.WriteLine("You have chosen to the end the program")
            End Select
            choice = menu()

        End While

    End Sub
    Sub load(ByRef schools() As String, ByRef wins() As Integer, ByVal filename As String)
        FileOpen(1, filename, OpenMode.Input)
        For index = 0 To maxArray
            Input(1, schools(index))
            Input(1, wins(index))
        Next
        FileClose(1)
        MsgBox("File read")
    End Sub
    Sub save(ByVal schools() As String, ByVal wins() As Integer, ByVal filename As String)
        FileOpen(1, filename, OpenMode.Output)
        For index = 0 To maxArray
            Write(1, schools(index))
            Write(1, wins(index))
        Next
        FileClose(1)
        MsgBox("File written")
    End Sub
    Sub displayData(ByVal schools() As String, ByVal wins() As Integer)
        For index = 0 To maxArray
            Console.WriteLine("School: " & schools(index) & vbTab & " Number of wins: " & vbTab & wins(index))
        Next
    End Sub
    Sub getData(ByRef schools() As String, ByRef wins() As Integer)
        Dim valid As Boolean = False
        For index = 0 To maxArray
            schools(index) = InputBox("Enter school name: ")
            valid = False
            While valid = False
                Try
                    wins(index) = InputBox("Enter wins 0 to 20")
                    While wins(index) < 0 Or wins(index) > 20
                        wins(index) = InputBox("Error. Enter wins 0 to 20")
                    End While
                    valid = True

                Catch ex As Exception
                    MsgBox("Error. Enter an integer")
                End Try
            End While
        Next
    End Sub
    Function menu() As Integer
        Dim response As Integer

        Console.WriteLine("Select from the following:")
        Console.WriteLine("1 Load data")
        Console.WriteLine("2 Enter new data")
        Console.WriteLine("3 Save data")
        Console.WriteLine("4 Display data")
        Console.WriteLine("5 Menu options")
        Console.WriteLine("6 Quit")

        response = Console.ReadLine

        While response < 1 Or response > 6
            Console.WriteLine("Error enter number 1 to 6")
            response = Console.ReadLine
        End While
        Return response
    End Function

End Module













